openhantek/src/scopesettings.h

```diff
@@ -55,14 +55,19 @@ struct DsoSettingsScopeVoltage {
     double offset = 0.0;        ///< Vertical offset in divs
     double trigger = 0.0;       ///< Trigger level in V
     unsigned gainStepIndex = 6; ///< The vertical resolution in V/div (default = 1.0)
-    union {                     ///< Different enums, coupling for real- and mode for math-channels
-        Dso::MathMode math;
-        unsigned couplingIndex = 0;
-        int rawValue;
-    };
-    QString name;          ///< Name of this channel
-    bool inverted = false; ///< true if the channel is inverted (mirrored on cross-axis)
-    bool used = false;     ///< true if this channel is enabled
+    bool inverted = false;      ///< true if the channel is inverted (mirrored on cross-axis)
+                   ///< Different enums, coupling for real- and mode for math-channels
+    Dso::MathMode math;
+    unsigned couplingIndex = 0;
+    int rawValue;
+
+    std::vector<double> probeGainSteps; ///< Probe gain steps for channel
+    double probe_gain = 1; ///< Default value for the probe gain
```

Comment:
Isn't it enough to have `probeStepIndex` and `defaultValues`?

Reply:
This was because this was used before directly on the raw data and i would not have to pay the cost for the array access.